GitUI/AutoCompletion/CommitAutoCompleteProvider.cs

```diff
@@ -34,13 +37,15 @@ public CommitAutoCompleteProvider(GitModule module)
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                var regex = GetRegexForExtension(Path.GetExtension(file.Name));
+                var extension = Path.GetExtension(file.Name);
+
+                var regexByExtension = await _regexByExtension.GetValueAsync(cancellationToken);
```

Comment:
does this need to be async?

Reply:
Previous code parsed file synchronously during the first call to `Lazy<Dictionary<...>>.Value`. That happend during show of `FormCommit`, which would block during IO, reading the file.

This change uses `AsyncLazy` which allows the factory function to be async, which allows async IO.

Reply:
we can probably initiate this process when we start the app, and thus won't need such complex structure, can probably replace it with `ConcurrentDictionary`...
thoughts?

Reply:
Could do. It's not a big saving here anyway as the file is small and fast to read. On the order of a few milliseconds in my testing with an SSD. I mostly wanted to experiment a little more with vs-threading.

Reply:
I'm happy to drop the second commit from this PR.

Reply:
KISS, if we can help it